app/services/idv/data_url_image.rb

```diff
@@ -1,16 +1,32 @@
 module Idv
   class DataUrlImage
     def initialize(data_url)
-      @header, base64_data = data_url.split(',', 2)
-      @data = Base64.decode64(base64_data || '')
+      header, data = URI(data_url.chomp).opaque.to_s.split(',', 2)
+      @header = header.to_s
+      @data = data.to_s
     end
 
+    BASE64_CONTENT_TYPE = /;base64$/.freeze
+
+    # @return [String]
     def content_type
-      @header.gsub(/^data:/, '').gsub(/;base64$/, '')
+      content_type, *_rest = @header.split(';')
+      content_type.to_s
     end
 
+    # @return [String]
     def read
-      @data
+      if base64_encoded?
+        Base64.decode64(@data)
+      else
+        URI.decode_www_form_component(@data)
```

Comment:
Following-up from discussion at https://github.com/18F/identity-idp/pull/4069#discussion_r470804449 :

Documented at : https://ruby-doc.org/stdlib-2.6/libdoc/uri/rdoc/URI.html#method-c-decode_www_form_component

>This decodes + to SP.

Seems to bring us back to the original reason for my comment 😄  What was it that Rubocop dislikes about `URI.decode`?

Reply:
Here's the cop: https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/UriEscapeUnescape, I just blindly followed it but clearly I should not have


```ruby
 [ URI.method(:decode), CGI.method(:unescape), URI.method(:decode_www_form), URI.method(:decod
_www_form_component) ].map { |m| m.call('a%20+%20b') }
=> ["a + b", "a   b", [["a   b", ""]], "a   b"]
```

| input | output | matches expectation |
| ---- | ---- |-----|
| `URI.decode('a%20+%20b')` | `"a + b"` | ✅ |
| `CGI.unescape('a%20+%20b')` |  `"a   b"` | ❌ |
| `URI.decode_www_form('a%20+%20b')` | `[["a   b", ""]]` |❌ |
| `URI.decode_www_form_component('a%20+%20b')` | `"a   b"` |❌ |



Fixed in 9e6c615e3 and rubocop ignored it